python/pylibcudf/pylibcudf/tests/io/test_csv.py

```diff
@@ -282,3 +283,109 @@ def test_read_csv_header(csv_table_data, source_or_sink, header):
 # list true_values = None,
 # list false_values = None,
 # bool dayfirst = False,
+
+
+def post_process_str_result(
+    str_result, sep=",", lineterminator="\n", header=True
+):
+    lines = str_result.split(lineterminator)[:-1]
+
+    if header:
+        result_lines = [
+            f"{i-1}{sep}{s}" if i > 0 else f"{sep}{s}"
+            for i, s in enumerate(lines)
+        ]
+    else:
+        result_lines = [f"{i}{sep}{s}" for i, s in enumerate(lines)]
+
+    str_result = lineterminator.join(result_lines)
+
+    if header or len(lines) > 1:
+        str_result += lineterminator
+
+    return str_result
+
+
+@pytest.mark.parametrize("sep", [",", "*"])
+@pytest.mark.parametrize("lineterminator", ["\n", "\n\n"])
+@pytest.mark.parametrize("header", [True, False])
+@pytest.mark.parametrize("rows_per_chunk", [8, 100])
+def test_write_csv(
+    table_data_with_non_nested_pa_types,
+    source_or_sink,
+    sep,
+    lineterminator,
+    header,
+    rows_per_chunk,
+):
+    plc_tbl_w_meta, pa_table = table_data_with_non_nested_pa_types
+    sink = source_or_sink
+
+    plc.io.csv.write_csv(
+        (
+            plc.io.csv.CsvWriterOptions.builder(
+                plc.io.SinkInfo([sink]), plc_tbl_w_meta.tbl
+            )
+            .names(plc_tbl_w_meta.column_names())
+            .na_rep("")
+            .include_header(header)
+            .rows_per_chunk(rows_per_chunk)
+            .line_terminator(lineterminator)
+            .inter_column_delimiter(sep)
+            .true_value("True")
+            .false_value("False")
+            .build()
+        )
+    )
+
+    # Convert everything to string to make comparisons easier
+    str_result = sink_to_str(sink)
+    str_result = post_process_str_result(
+        str_result, sep=sep, lineterminator=lineterminator, header=header
+    )
```

Comment:
Why do we need to post-process? What is different between this CSV (before post-processing) and the one that pandas writes?

Reply:
The str result from before post processing does not have indices like. I had to add them afterwards.
0 ...
1 ...
2 ...
3 ...
...

Reply:
Sure thing